application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/ConfigurationDataProvider.java

```diff
@@ -0,0 +1,180 @@
+public class ConfigurationDataProvider extends AbstractDataProvider
+{
+    /**
+     * The hint for the component.
+     */
+    public static final String HINT = "configuration";
+
+    private final String template = "data/configurationTemplate.vm";
+
+    private Map<String, String> supportedDB;
+
+    @Inject
+    private DefaultFileOperations fileOperations;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        supportedDB = new HashMap<>();
+        supportedDB.put("mysql", "MySQL");
+        supportedDB.put("hsqldb", "HSQLDB");
+        supportedDB.put("mariadb", "MariaDB");
+        supportedDB.put("postgresql", "PostgreSQL");
+        supportedDB.put("oracle", "Oracle");
+    }
+
+    @Override
+    public String getIdentifier()
+    {
+        return HINT;
+    }
+
+    @Override
+    public String provideData()
+    {
+        Map<String, String> systemInfo = generateJson();
+        if (systemInfo != null) {
+            systemInfo.put(serverFound, "found");
+        } else {
+            systemInfo = new HashMap<>();
+            systemInfo.put(serverFound, null);
+            systemInfo.put("supportedServers", currentServer.getSupportedServers().toString());
+        }
+        return renderTemplate(template, systemInfo, HINT);
+    }
+
+    @Override
+    public Map<String, String> generateJson()
+    {
+        try {
+            currentServer.updateCurrentServer();
+            Map<String, String> systemInfo = new HashMap<>();
+            systemInfo.put("xwikiCfgPath", currentServer.getCurrentServer().getXwikiCfgFolderPath());
+            systemInfo.put("tomcatConfPath", this.currentServer.getCurrentServer().getServerCfgPath());
+            systemInfo.put("javaVersion", this.getJavaVersion());
+            systemInfo.putAll(this.getOSInfo());
+            systemInfo.put("database", this.identifyDB());
+            systemInfo.put("usedServer", this.currentServer.getCurrentServer().getComponentHint());
+            return systemInfo;
+        } catch (Exception e) {
+            logger.warn("Failed to generate the configuration details. Error info : [{}]",
+                ExceptionUtils.getRootCauseMessage(e));
+            return null;
+        }
+    }
+
+    /**
+     * Get the version of Java used on the server.
+     *
+     * @return the used Java version.
+     */
+    String getJavaVersion()
+    {
+        return System.getProperty("java.version");
+    }
+
+    /**
+     * Identify the used database for XWiki by verifying the configration files.
+     *
+     * @return the name of the used database.
+     */
+    String identifyDB()
+    {
+        String databaseCfgPath = currentServer.getCurrentServer().getXwikiCfgFolderPath() + "hibernate.cfg.xml";
+        File file = new File(databaseCfgPath);
+
+        try {
+            fileOperations.initializeScanner(file);
+            String usedDB = "Database not found";
```

Comment:
You could initialize it outside of the try section, and just return usedDB at the end. So in case something goes wrong, you just not that this is what it's returned. Or there a different between the "Database not found" and the null value returned?